auth/src/main/java/org/aerogear/auth/utils/AuthStateManager.java

```diff
@@ -39,11 +39,11 @@ public OIDCCredentials read() {
      * Saves a token
      * @param authState token to be saved
      */
-    public synchronized void write(final OIDCCredentials authState) {
+    public synchronized void write(final OIDCCredentials authState) throws JSONException {
         if (authState == null) {
             clear();
         } else {
-            if(!prefs.edit().putString(KEY_STATE, authState.serialise()).commit()) {
+            if(!prefs.edit().putString(KEY_STATE, authState.serialize()).commit()) {
                 throw new IllegalStateException("Failed to update state from shared preferences");
```

Comment:
Is it correct to throw an unchecked exception here?

Reply:
@ziccardi Yes.

Reply:
However, we should add `@throws`  to the javadocs